File: Data Structures & Algorithms/non-overlapping-intervals/submission-8.py
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        intervals.sort(key = lambda x:x[0])
        last = None
        res = 0
        for start,end in intervals:
            if not last:
                last = end
                continue
            if last>start:
                res +=1
                last = min(last,end)
            else:
                last = end
        return res
            
        

File: Data Structures & Algorithms/meeting-schedule-ii/submission-8.java
/**
 * Definition of Interval:
 * public class Interval {
 *     public int start, end;
 *     public Interval(int start, int end) {
 *         this.start = start;
 *         this.end = end;
 *     }
 * }
 */

class Solution {
    public int minMeetingRooms(List<Interval> intervals) {
        
        intervals.sort((a,b) ->a.start-b.start);
        PriorityQueue<Integer> minHeap = new PriorityQueue<>();
        int res = 0;
        for(Interval interval:intervals){
            minHeap.offer(interval.end);
            while(!minHeap.isEmpty() && minHeap.peek()<=interval.start){
                minHeap.poll();
            }
            res = Math.max(res,minHeap.size());
        }
        return res;

    }
}
